### ISSUE TITLE
deprecation warnings on py312

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#7860]
1.2-maint branch:
```
    bookworm64: .tox/py312-fuse3/lib/python3.12/site-packages/dateutil/tz/tz.py:37: 16 warnings
    bookworm64:   /vagrant/borg/borg/.tox/py312-fuse3/lib/python3.12/site-packages/dateutil/tz/tz.py:37: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    bookworm64:     EPOCH = datetime.datetime.utcfromtimestamp(0)
    bookworm64: 

    bookworm64: .tox/py312-fuse3/lib/python3.12/site-packages/borg/testsuite/archiver.py::ArchiverTestCaseBinary::test_migrate_lock_alive
    bookworm64:   /vagrant/borg/borg/.tox/py312-fuse3/lib/python3.12/site-packages/borg/testsuite/__init__.py:265: DeprecationWarning: This process (pid=187224) is multi-threaded, use of fork() may lead to deadlocks in the child.
    bookworm64:     if os.fork() == 0:
    bookworm64: 

    bookworm64: .tox/py312-fuse3/lib/python3.12/site-packages/borg/testsuite/helpers.py::test_safe_timestamps
    bookworm64:   /vagrant/borg/borg/.tox/py312-fuse3/lib/python3.12/site-packages/borg/testsuite/helpers.py:1098: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    bookworm64:     datetime.utcfromtimestamp(beyond_y10k)
    bookworm64: 

    bookworm64: .tox/py312-fuse3/lib/python3.12/site-packages/borg/testsuite/helpers.py::test_safe_timestamps
    bookworm64:   /vagrant/borg/borg/.tox/py312-fuse3/lib/python3.12/site-packages/borg/testsuite/helpers.py:1099: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    bookworm64:     assert datetime.utcfromtimestamp(safe_s(beyond_y10k)) > datetime(2038, 1, 1)
    bookworm64: 

    bookworm64: .tox/py312-fuse3/lib/python3.12/site-packages/borg/testsuite/helpers.py::test_safe_timestamps
    bookworm64:   /vagrant/borg/borg/.tox/py312-fuse3/lib/python3.12/site-packages/borg/testsuite/helpers.py:1100: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    bookworm64:     assert datetime.utcfromtimestamp(safe_ns(beyond_y10k) / 1000000000) > datetime(2038, 1, 1)
```

**Comment by ThomasWaldmann** [comment]
Related: https://github.com/borgbackup/borg/pull/7715 (master branch)

**Comment by ThomasWaldmann** [comment]
Fixed by #7857.
